org-roam-db.el

```diff
@@ -332,8 +332,8 @@ connections, nil is returned."
 
 (defun org-roam-db--update-tags ()
   "Update the tags of the current buffer into the cache."
-  (let* ((file (file-truename (buffer-file-name)))
-         (tags (org-roam--extract-tags)))
+  (when-let ((file (file-truename (buffer-file-name)))
+             (tags (org-roam--extract-tags)))
```

Comment:
I don't think this will behave correctly when a file originally has a tag, and then they are later removed.

`org-roam--extract-tags` will return nil and then the entry will not be deleted from the db.

I think the db insert should be guarded by the when, but not the delete.

e.g.
`
  (let ((file (file-truename (buffer-file-name)))
        (tags (org-roam--extract-tags)))
    (org-roam-db-query [:delete :from tags
                        :where (= file $s1)]
                       file)
    (when tags (org-roam-db--insert-tags file tags))))
`

Reply:
Yes you are right. I didn’t realize that.